minerl/herobraine/hero/handlers/agent/observations/location_stats.py

```diff
@@ -0,0 +1,158 @@
+
+class _SkyLightLevelObservation(_FullStatsObservation):
+    def __init__(self):
+        super().__init__(key_list=['sky_light_level'], space=spaces.Box(low=0.0, high=1.0, shape=(), dtype=np.float), default_if_missing=0.71)
+
+class _XPositionObservation(_FullStatsObservation):
+    def __init__(self):
+        super().__init__(key_list=['xpos'], space=spaces.Box(low=-640000.0, high=640000.0, shape=(), dtype=np.float), default_if_missing=0.0)
+class _YPositionObservation(_FullStatsObservation):
+    def __init__(self):
+        super().__init__(key_list=['ypos'], space=spaces.Box(low=-640000.0, high=640000.0, shape=(), dtype=np.float), default_if_missing=0.0)
+class _ZPositionObservation(_FullStatsObservation):
+    def __init__(self):
+        super().__init__(key_list=['zpos'], space=spaces.Box(low=-640000.0, high=640000.0, shape=(), dtype=np.float), default_if_missing=0.0)
+
+class _PitchObservation(_FullStatsObservation):
+    def __init__(self):
+        super().__init__(key_list=['pitch'], space=spaces.Box(low=-180.0, high=180.0, shape=(), dtype=np.float),
+                         default_if_missing=0.0)
+
+class _YawObservation(_FullStatsObservation):
+    def __init__(self):
+        super().__init__(key_list=['yaw'], space=spaces.Box(low=-180.0, high=180.0, shape=(), dtype=np.float),
+                         default_if_missing=0.0)
+
+class _BiomeIDObservation(_FullStatsObservation):
+    def to_hero(self, x):
+        for key in self.hero_keys:
+            x = x[key]
+        return np.eye(40)[x]
+
+    def __init__(self):
+        super().__init__(key_list=['biome_id'],
+                         space=spaces.Discrete(40),
+                         default_if_missing=np.eye(40)[0])
+
+class _BiomeRainfallObservation(_FullStatsObservation):
+    def __init__(self):
+        super().__init__(key_list=['biome_rainfall'],
+                         space=spaces.Box(low=0.0, high=1.0, shape=()),
+                         default_if_missing=0.5)
+
+class _BiomeTemperatureObservation(_FullStatsObservation):
+    def __init__(self):
+        super().__init__(key_list=['biome_temperature'],
+                         space=spaces.Box(low=0.0, high=1.0, shape=()),
+                         default_if_missing=0.5)
+
+class _SeaLevelObservation(_FullStatsObservation):
+    def __init__(self):
+        super().__init__(key_list=['sea_level'],
+                         space=spaces.Box(low=0.0, high=255, shape=(), dtype=np.int),
+                         default_if_missing=63)
+
+class _LightLevelObservation(_FullStatsObservation):
+    def __init__(self):
+        super().__init__(key_list=['light_level'],
+                         space=spaces.Box(low=0.0, high=15, shape=(), dtype=np.int),
+                         default_if_missing=15)
+
+class _IsRainingObservation(_FullStatsObservation):
+    def to_hero(self, x):
+        for key in self.hero_keys:
+            x = x[key]
+        return np.eye(2)[x]
```

Comment:
For these boolean observations, couldn't we just use a single element that could be 0 or 1 instead of one-hot encoding it?

Reply:
Yep!

Reply:
Does that mean you're going to change it?